src/pybind/cephfs/setup.py

```diff
@@ -9,7 +9,32 @@
 import textwrap
 from distutils.ccompiler import new_compiler
 from distutils.errors import CompileError, LinkError
+import distutils.sysconfig
 from distutils.sysconfig import customize_compiler
+unwrapped_customize = distutils.sysconfig.customize_compiler
+
+clang = False
+
+def monkey_with_compiler(compiler):
+    unwrapped_customize(compiler)
+    if (compiler.compiler_type == 'unix'):
+        cmd = compiler.compiler
+        if cmd[0].find('clang') != -1:
+            global clang
+            clang = True
+            cmd = [c for c in cmd if not (c == '-mcet' or
+                                          c.startswith('-fcf-protection'))]
+            cmd_so = compiler.compiler_so
+            cmd_so = [c for c in cmd_so if not
+                      (c == '-mcet' or c.startswith('-fcf-protection'))]
+            compiler.compiler = cmd
+            compiler.set_executable('compiler', ' '.join(cmd))
+            compiler.compiler_so = cmd_so
+            compiler.set_executable('compiler_so', ' '.join(cmd_so))
+
+distutils.sysconfig.customize_compiler = monkey_with_compiler
+customize_compiler = monkey_with_compiler
```

Comment:
Because we import cusomtize compiler specifically, if we don't set both we still get the old one.

THOUGH I'll just change it so when we call customize compiler we do it fully qualified and eliminate the spare.